Makefile

```diff
@@ -17,28 +17,53 @@
 #
 # Makefile with some common workflow for dev, build and test
 #
+export GOPROXY?=https://proxy.golang.org/
+CONTROLLER_GEN_BIN_PATH := $(shell which controller-gen)
+.DEFAULT_GOAL:=help
 
-all: build test
+##@ General
 
-.PHONY: build test
+.PHONY: help
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```

Comment:
heh.  Please comment this for posterity -- something like 

```makefile
# print out usage with ansi control characters for terminal formatting.
# see https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters for more info
# the awk does commands do ...
```

AFAICT, the awk commands read the entire set of makefiles included in this invocation, looking for lines of the form `xyz: ## something`, and then pretty-format the target and help.  If there's a line `##@ something`, that gets pretty-printed as a heading.